src/AssignTransfer/components/OffenderSearch.js

```diff
@@ -5,48 +5,72 @@ import { withRouter } from 'react-router';
       <div>
-        <div className="pure-u-md-12-12 searchForm">
-          <div className="padding-top padding-left padding-right">
-            <label className="form-label" htmlFor="seachText">Offender name or number</label>
-            <input type="text" className="form-control" id="search-text" name="searchText" value={this.props.searchText} onChange={this.props.handleSearchTextChange}/>
-            <button className="button margin-left" onClick={() => this.props.gotoNext(this.props.history)}>Search ></button>
+        <label className="form-label" htmlFor="housing-location-select">Housing location</label>
+        <select id="housing-location-select" name="housing-location-select" className="form-control"
+          value={this.props.housingLocation}
+          onChange={this.props.handleSearchHousingLocationChange}>
+          <option key="choose" value="">-- Select --</option>
+          {housingLocations}
+        </select></div>);
+
+    if (this.props.singleLineLayout) {
+      return (
+        <div>
+          <div className="pure-u-md-12-12 searchForm padding-bottom">
+            <div className="pure-u-md-4-12 padding-top padding-left">
+              <label className="form-label" htmlFor="seachText">Offender name or number</label>
+              <input type="text" className="form-control width100" id="search-text" name="searchText"
+                value={this.props.searchText} onChange={this.props.handleSearchTextChange}/>
+            </div>
+            <div className="pure-u-md-5-12 padding-top padding-left">
+              {locationSelect}
+            </div>
+            <div className="pure-u-md-2-12 padding-top padding-left">
+              <label className="form-label">&nbsp;</label>
+              <button className="button" onClick={() => this.props.doSearch()}>Search again</button>
+            </div>
           </div>
-          <div className="padding-top padding-left padding-right padding-bottom-large">
-            <div className="pure-u-md-5-12 padding-right">
-              <label className="form-label" htmlFor="housing-location-select">Housing location</label>
-              <select id="housing-location-select" name="housing-location-select" className="form-control" value={this.props.housingLocation}
-                onChange={this.props.handleSearchHousingLocationChange}>
-                <option key="choose" value="--">-- Select --</option>
-                {housingLocations}
-              </select>
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <div className="pure-u-md-12-12 searchForm">
+            <div className="padding-top padding-left padding-right">
+              <label className="form-label" htmlFor="seachText">Offender name or number</label>
+              <input type="text" className="form-control width70 margin-bottom" id="search-text" name="searchText"
+                value={this.props.searchText} onChange={this.props.handleSearchTextChange}/>
```

Comment:
Im not keen on having 2 different form submit methods . I couldnt use a push when already on the results page because then nothing happens!